src/compiler/checker.ts

```diff
@@ -5299,7 +5299,7 @@ namespace ts {
          * Check if a Type was written as a tuple type literal.
          * Prefer using isTupleLikeType() unless the use of `elementTypes` is required.
          */
-        function isTupleType(type: Type): boolean {
+        function isTupleType(type: Type): type is TupleType {
             return (type.flags & TypeFlags.Tuple) && !!(<TupleType>type).elementTypes;

```

Comment:
Yeah I know, but createTupleType always sets the .elementTypes property, so we can eliminate that property check.